### ISSUE TITLE
Spreading complex numbers

### ISSUE THREAD

**Issue by krlmlr** [hadley/tidyr#134]
...gives garbage output (at least on my system, Ubuntu).

``` r
library(tidyr)
```

    ## 
    ## Attaching package: 'tidyr'
    ## 
    ## The following object is masked from 'package:magrittr':
    ## 
    ##     extract

``` r
library(dplyr)
```

    ## 
    ## Attaching package: 'dplyr'
    ## 
    ## The following objects are masked from 'package:stats':
    ## 
    ##     filter, lag
    ## 
    ## The following objects are masked from 'package:base':
    ## 
    ##     intersect, setdiff, setequal, union

``` r
data <-
  expand.grid(id = 1:3, key = letters[1:2]) %>%
  mutate(value = complex(real = runif(6), imag = runif(6)))
data
```

    ##   id key                value
    ## 1  1   a 0.9361484+0.1026268i
    ## 2  2   a 0.2283434+0.3072617i
    ## 3  3   a 0.7494127+0.8295208i
    ## 4  1   b 0.6885753+0.2452242i
    ## 5  2   b 0.5677571+0.2666375i
    ## 6  3   b 0.7047463+0.4077817i

``` r
data %>% spread(key, value)
```

    ##   id        a    b
    ## 1  1 NaN+NaNi 0+0i
    ## 2  2 NaN+NaNi 0+0i
    ## 3  3 NaN+  0i 0+0i

``` r
devtools::session_info()
```

    ## Session info --------------------------------------------------------------

    ##  setting  value                       
    ##  version  R version 3.2.2 (2015-08-14)
    ##  system   x86_64, linux-gnu           
    ##  ui       RStudio (0.99.486)          
    ##  language en_US:en                    
    ##  collate  en_US.UTF-8                 
    ##  tz       <NA>                        
    ##  date     2015-10-30

    ## Packages ------------------------------------------------------------------

    ##  package    * version     date       source        
    ##  assertthat   0.1         2013-12-06 CRAN (R 3.2.0)
    ##  DBI          0.3.1.9008  2015-10-27 local         
    ##  devtools     1.9.1.9000  2015-10-21 local         
    ##  digest       0.6.8       2014-12-31 CRAN (R 3.2.0)
    ##  dplyr      * 0.4.3       2015-09-01 CRAN (R 3.2.2)
    ##  evaluate     0.8.1       2015-10-06 local         
    ##  htmltools    0.2.6       2014-09-08 CRAN (R 3.2.0)
    ##  knitr        1.11        2015-08-14 CRAN (R 3.2.1)
    ##  lazyeval     0.1.10.9000 2015-08-21 local         
    ##  magrittr   * 1.5         2015-10-19 local         
    ##  memoise      0.2.99.9000 2015-10-08 local         
    ##  R6           2.1.1       2015-09-19 local         
    ##  Rcpp         0.12.1      2015-09-10 CRAN (R 3.2.2)
    ##  rmarkdown    0.8.1       2015-10-10 CRAN (R 3.2.2)
    ##  stringi      1.0-1       2015-10-22 CRAN (R 3.2.2)
    ##  stringr      1.0.0       2015-04-30 CRAN (R 3.2.0)
    ##  tidyr      * 0.3.1.9000  2015-10-30 local         
    ##  ulimit       0.0-2       2015-04-14 local

**Comment by krlmlr** [comment]
Only with `convert = FALSE`.

**Comment by krlmlr** [comment]
Will be fixed by #131 + hadley/dplyr#1493.
